### PR TITLE
Add properties to affect text & background colors.

### PR DESCRIPTION
**Pull request by levigroker**
Adds simple text color and background color properties to aid in customization.

### CODE REVIEW

File: Classes/VTAcknowledgementsViewController.m [hunk 1]
```diff
@@ -232,7 +234,7 @@ - (void)configureHeaderView
     UILabel *label = [[UILabel alloc] initWithFrame:labelFrame];
     label.text             = self.headerText;
     label.font             = font;
-    label.textColor        = [UIColor grayColor];
+	label.textColor        = self.textColor ?: [UIColor grayColor];
```
Comment (moritzmoritz) [inline Classes/VTAcknowledgementsViewController.m]: Remove Tab :+1:

File: Classes/VTAcknowledgementsViewController.m [hunk 2]
```diff
@@ -255,7 +257,7 @@ - (void)configureFooterView
     UILabel *label = [[UILabel alloc] initWithFrame:labelFrame];
     label.text             = self.footerText;
     label.font             = font;
-    label.textColor        = [UIColor grayColor];
+	label.textColor        = self.textColor ?: [UIColor grayColor];
```
Comment (moritzmoritz) [inline Classes/VTAcknowledgementsViewController.m]: See above :+1:

File: Classes/VTAcknowledgementsViewController.m [hunk 3]
```diff
@@ -394,6 +396,11 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     VTAcknowledgement *acknowledgement = self.acknowledgements[indexPath.row];
     cell.textLabel.text = acknowledgement.title;
     cell.accessoryType  = UITableViewCellAccessoryDisclosureIndicator;
+	cell.textLabel.textColor = self.textColor;
+	cell.backgroundColor = self.cellBackgroundColor;
+	UIView *selectedBackgroundView = self.cellSelectedBackgroundColor ? [[UIView alloc] init] : nil;
```
Comment (moritzmoritz) [inline Classes/VTAcknowledgementsViewController.m]: a `[UIView new]`does the job

Comment (moritzmoritz) [inline Classes/VTAcknowledgementsViewController.m]: And again the tabs

File: Classes/VTAcknowledgementsViewController.m (hunk 1 above)
Comment (moritzmoritz) [inline Classes/VTAcknowledgementsViewController.m]: and ad the `?` right to the `textColor`

**Comment by vtourraine** [comment]
Thank you for the pull request Levi (and thank you Moritz for the review).

Unfortunately, I’m not really interested in adding this kind of customization properties. I totally understand how they can be useful, but I believe this library should remain as generic and as simple as possible, and let people customize it at the app level. Also the system `UIAppearance` proxy can sometimes be a better solution for this kind of problems. Does that seem reasonable?

**Comment by levigroker** [comment]
@vtourraine Sadly we were unable to get UIAppearance to modify the background color of the UITextVew, and UITableCell text. Using UIAppearance would be ideal, of course, but it was inoperative for the VTAcknowledgements UI for unknown reasons.

The rationale behind these modifications is to allow customization for a "dark" theme, so while I appreciate wanting to keep the code simple, allowing clients to customize the appearance is important for libraries/frameworks which offer a UI.

I hope you will reconsider.

File: Classes/VTAcknowledgementsViewController.m (hunk 1 above)
Comment (levigroker) [inline Classes/VTAcknowledgementsViewController.m]: @moritzmoritz Thanks... will convert tabs to spaces, as you like. I don't understand your other comment about the `?`, however. Can you clarify please?

File: Classes/VTAcknowledgementsViewController.m (hunk 3 above)
Comment (levigroker) [inline Classes/VTAcknowledgementsViewController.m]: @moritzmoritz Happy to use `new` if you prefer. My sources imply preferring alloc/init as the standard, however: https://twitter.com/bbum/status/834105958259253248

**Comment by vtourraine** [comment]
The dark mode option is interesting, because that’s actually a system feature now ([UIUserInterfaceStyle](https://developer.apple.com/reference/uikit/uiuserinterfacestyle)). It's currently limited to tvOS, but iOS 11 might change that.

**Comment by vtourraine** [comment]
Also worth noting: the dark mode in tvOS is supported automatically when using system UI elements (e.g. if you keep the default text color and the default background color).

Again, I’m not saying that these customization properties are useless, I just believe that this library should remain basic (and leave the door open to subclass/customization).

**Comment by levigroker** [comment]
Certainly your prerogative.
May I suggest, at a minimum, the colors for the header/footer labels are not hard-coded?
Sadly, this means I'll have to use my own fork until such time we see improvements which allow us to style the exposed UI.

**Comment by vtourraine** [comment]
> May I suggest, at a minimum, the colors for the header/footer labels are not hard-coded?

That would be great actually, but I’m not sure that can be done with UIKit as it is, except leaving the text to the default black (but that doesn’t look great). We have `[UIColor darkTextColor]` and `[UIColor lightTextColor]`, but no middle ground for this case.

**Pull request closed without merging**